app/controllers/api/v1/cars_controller.rb

```diff
@@ -1,43 +1,82 @@
 class Api::V1::CarsController < ApplicationController
+  # api/v1/cars
   def index
-    @cars = Car.all
-    render json: @cars
+    list_cars = Car.includes(:images).all.as_json(include: :images)
+    render json: {
+      data: {
+        cars: list_cars
+      }
+    }, status: :ok
   end
 
+  # api/v1/cars/{id}
   def show
-    @car = Car.find(params[:id])
-    render json: @car
+    car = Car.find_by(id: params[:id])
+    if car
+      render json: {
+        data: {
+          cars: Car.find_by(id: params[:id]).as_json(include: :images)
```

Comment:
- Although rails will cache the first query for a second use. It would be better if you run a single query to get the car record.
```ruby
    car = Car.find_by(id: params[:id])
    if car
      render json: {
        data: {
          cars: car.as_json(include: :images)
        }
      }, status: :ok
```